test/unit/spec/signaling/v2/peerconnection.js

```diff
@@ -5,12 +5,34 @@ const EventEmitter = require('events');
 const sinon = require('sinon');
 
 const EventTarget = require('../../../../../lib/eventtarget');
+const IceConnectionMonitor = require('../../../../../lib/signaling/v2/iceconnectionmonitor.js');
 const PeerConnectionV2 = require('../../../../../lib/signaling/v2/peerconnection');
 const { MediaClientLocalDescFailedError } = require('../../../../../lib/util/twilio-video-errors');
 const { FakeMediaStreamTrack } = require('../../../../lib/fakemediastream');
 const { a, combinationContext, makeEncodingParameters } = require('../../../../lib/util');
 
 describe('PeerConnectionV2', () => {
+  let didStartMonitor;
+  let didStopMonitor;
+  let inactiveCallback;
+  beforeEach(() => {
+    // stub out IceConnectionMonitor to not have any side effects
+    didStartMonitor = false;
+    didStopMonitor = false;
+    inactiveCallback = null;
+    sinon.stub(IceConnectionMonitor.prototype, 'start').callsFake((callback) => {
+      inactiveCallback = callback;
+      didStartMonitor = true;
+    });
+    sinon.stub(IceConnectionMonitor.prototype, 'stop').callsFake(() => {
+      didStopMonitor = true;
+    });
+  });
+  afterEach(() => {
+    IceConnectionMonitor.prototype.start.restore();
+    IceConnectionMonitor.prototype.stop.restore();
+  });
+
```

Comment:
@makarandp0 Can you also add or update the test for `PeerConnectionV2#iceConnectionState` to return "failed" if the actual state is "disconnected" and ICE is inactive, and should return the underlying RTCPeerConnection's `iceConnectionState` otherwise?

Reply:
oh, I did not think about this - Do we need to keep track of the ice failure predicted by monitor and pretend that it was real failure to the callers of PeerConnectionV2 ? Let me sync up with you  understand this better.